client/state/ui/oauth2-clients/reducer.js

```diff
@@ -15,11 +15,11 @@ import { ROUTE_SET } from 'state/action-types';
 export const currentClientId = createReducer( null, {
 	[ ROUTE_SET ]: ( state, { path, query } ) => {
 		if ( startsWith( path, '/log-in' ) ) {
-			return query.client_id || state;
+			return +query.client_id || state;
 		}
 
 		if ( startsWith( path, '/start/wpcc' ) ) {
-			return query.oauth2_client_id || state;
+			return +query.oauth2_client_id || state;
```

Comment:
Why didn't you use `Number()` to convert that string into an integer? This syntax looks like a recipe for bugs to me.

Reply:
It's a valid language [construct](https://stackoverflow.com/questions/17106681/parseint-vs-unary-plus-when-to-use-which/17106702#17106702)
Not sure why would you prefer `Number` here...

Reply:
Just for the reasons I mentioned above. When you look at this code you wonder if that's a typo, or if some part was just missing in front of this property. That's the type of hacks that makes people reading this code _think_, thus make this code less simpler to understand. I would make that casting explicit, especially as we've been using `Number()` in quite a few places already.

Reply:
It's a common pattern in JS to convert to number with unary `+`,
just like you convert usually date `+new Date()`